### PR TITLE
SRE-1904 Diagnostic test

### PR DESCRIPTION
**Pull request by JohnMalmberg**
Diagnostic test for SRE-1904

### CODE REVIEW

**Comment by JohnMalmberg** [comment]
This is needed to help diagnose SRE-1904 which only is showing up on timed or merges to to the release/2.4 branch.

File: Jenkinsfile [hunk 1]
```diff
@@ -124,6 +124,20 @@ pipeline {
                         }
                     }
                 }
+                stage('distroVersion() tests') {
+                    steps {
+                        withEnv(['BRANCH_NAME=release2.4']) {
```
Comment (brianjmurrell) [inline Jenkinsfile:129]: The branch name is `release/2.4`.

```suggestion
                        withEnv(['BRANCH_NAME=release/2.4']) {
```

It probably doesn't make a material difference to the test, but should be accurate at least for appearances sake.

File: Jenkinsfile
Comment (brianjmurrell) [inline Jenkinsfile:1]: Might add a Leap 15.4 test there to exercise the RE a bit more.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Do you have a Jenkins run with the fix removed to show that the tests catch the issue?

**Comment by JohnMalmberg** [comment]
Will remove the Jenkinsfile pipeline-lib self reference on the next push.

**Comment by brianjmurrell** [comment]
> failure_test: https://build.hpdd.intel.com/job/daos-stack/job/pipeline-lib/view/change-requests/job/PR-385/10/

This one reports the stage has no steps and has an error at the end of it:

```
[2023-07-06T15:04:23.221Z] java.lang.NullPointerException: Cannot invoke method startsWith() on null object
[2023-07-06T15:04:23.221Z] 	at org.codehaus.groovy.runtime.NullObject.invokeMethod(NullObject.java:91)
[2023-07-06T15:04:23.221Z] 	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:47)
[2023-07-06T15:04:23.221Z] 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
[2023-07-06T15:04:23.221Z] 	at org.codehaus.groovy.runtime.callsite.NullCallSite.call(NullCallSite.java:34)
[2023-07-06T15:04:23.221Z] 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
[2023-07-06T15:04:23.221Z] 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
[2023-07-06T15:04:23.221Z] 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:180)
[2023-07-06T15:04:23.221Z] 	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:182)
[2023-07-06T15:04:23.221Z] 	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
[2023-07-06T15:04:23.221Z] 	at WorkflowScript.run(WorkflowScript:132)
```

But indeed, why would it report no steps?  I guess ultimately that `script` is not a reportable step if it doesn't produce any output?

> Success test with leap15 added: https://build.hpdd.intel.com/job/daos-stack/job/pipeline-lib/view/change-requests/job/PR-385/12/

Again a very silent test.  Which when it passes, I guess is OK.  But when it fails, we need to know what failed.

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**